app/controllers/root_controller.rb

```diff
@@ -156,8 +156,11 @@ def publication
       end
 
       @interaction_details = prepare_interaction_details(@publication, authority_slug, snac)
-      if local_authority_match_but_no_interaction?(@interaction_details)
-        @location_error = error_for_missing_interaction(@interaction_details)
+      if local_authority_match?(@interaction_details)
+        @local_authority = LocalAuthorityPresenter.new(@interaction_details['local_authority'])
+        if local_authority_match_but_no_interaction?(@interaction_details)
+          @location_error = error_for_missing_interaction(@local_authority)
+        end
       end
```

Comment:
This code is new to me, so maybe I missed something, but is it necessary to check if local_authority_matches again, given it's inside the block that already did? 
```ruby
if local_authority_match?(@interaction_details)
  @local_authority = LocalAuthorityPresenter.new(@interaction_details['local_authority'])
  if no_interaction?(@interaction_details)
    @location_error = error_for_missing_interaction(@local_authority)
  end
end
```